cmd/osmosisd/cmd/root.go

```diff
@@ -47,6 +54,146 @@ import (
 	osmosis "github.com/osmosis-labs/osmosis/v17/app"
 )
 
+type Trace struct {
+	Type         string `json:"type"`
+	Counterparty struct {
+		BaseDenom string `json:"base_denom"`
+	} `json:"counterparty"`
+}
+
+type Asset struct {
+	Display string  `json:"display"`
+	Base    string  `json:"base"`
+	Traces  []Trace `json:"traces"`
+}
+
+type AssetList struct {
+	Assets []Asset `json:"assets"`
+}
+
+func loadAssetList(initClientCtx client.Context, cmd *cobra.Command, udenomToIBC, IBCtoUdenom bool) (map[string]string, map[string]string) {
+	var assetListURL string
+	var assetList AssetList
+
+	chainId := GetChainId(initClientCtx, cmd)
+
+	if chainId == "osmosis-1" || chainId == "" {
+		assetListURL = "https://raw.githubusercontent.com/osmosis-labs/assetlists/main/osmosis-1/osmosis-1.assetlist.json"
+	} else if chainId == "osmo-test-5" {
+		assetListURL = "https://raw.githubusercontent.com/osmosis-labs/assetlists/main/osmo-test-5/osmo-test-5.assetlist.json"
+	} else {
+		return nil, nil
+	}
```

Comment:
This is already a big win! But what I think would make this better is if we kept the downloaded file locally, so we don't have network delay on each of these calls. We can perhaps spin up a second thread to check for updates?

Reply:
So is your recommendation that on the first call we download the file and store to the .osmosisd folder, and then subsequent calls check for the existence of this file?

Reply:
Oh maybe you answered this with go embed, let me take a look and then come back to this question i asked

Reply:
go embed is for making the default assetlist compiled with the binary

Reply:
What i was thinking though is that adding a network delay for downloading a file for every CLI action will be a notable delay.

Reply:
Oh maybe we can handle updating the file in another command altogether. (E.g. `osmosisd config update`, and then we can bundle in more recommendations between versions that we want people to do there)

So then the flow for here would be:
"If local file, use that. Else use go-embed file"

And a separate command automatically downloads the local files.

Reply:
Okay so to be clear:

* We have a file that is embedded in the binary that never gets updated (maybe each release it gets updated)
* We check for the local file, which can be updated with a separate cli command. 
* If this local file doesnt exist, we fall back to the embedded file (the one that doesn't get updated)

Reply:
Yup!

Reply:
Awesome, agreed, will implement!